megamek/src/megamek/common/weapons/WeaponHandler.java

```diff
@@ -1479,17 +1479,27 @@ protected void handleEntityDamage(Entity entityTarget,
         missed = false;
 
         initHit(entityTarget);
+        hit.setLocation(Mech.LOC_LLEG);
         
         boolean isIndirect = wtype.hasModes()
                 && weapon.curMode().equals("Indirect");
         IHex targetHex = game.getBoard().getHex(target.getPosition());
+        // a specific situation where a mech is standing in depth 1 water
+        boolean mechPokingOutOfShallowWater = targetHex.containsTerrain(Terrains.WATER) &&
+                entityTarget.relHeight() <= targetHex.surface();
```

Comment:
Why `<=` instead of `>=`?

Reply:
Actually, now that I think about it, it should be == only. We don't want >=, because stuff like superheavy mechs (technically height 3) shouldn't get partial cover benefits from standing in level 1 water.